core/src/main/scala/com/avast/cloud/metrics/datadog/api/Gauge.scala

```diff
@@ -0,0 +1,8 @@
+package com.avast.cloud.metrics.datadog.api
+
+import scala.language.higherKinds
+
+trait Gauge[F[_]] {
+  def set(value: Long, tags: Tag*): F[Unit]
```

Comment:
I don't really like method overloading as it makes type inference difficult. How about parametrizing `Gauge ` into something like `Gauge[F[_],T]`